lucene/core/src/test/org/apache/lucene/search/TestDocValuesFieldExistsQuery.java

```diff
@@ -206,6 +210,50 @@ public void testFieldExistsButNoDocsHaveField() throws IOException {
     dir.close();
   }
 
+  public void testQueryMatchesCount() throws IOException {
+    Directory dir = newDirectory();
+    RandomIndexWriter w = new RandomIndexWriter(random(), dir);
+
+    int randomNumDocs = TestUtil.nextInt(random(), 10, 100);
+    int numMatchingDocs = 0;
+
+    for (int i = 0; i < randomNumDocs; i++) {
+      Document doc = new Document();
+      if (random().nextBoolean()) {
+        doc.add(new LongPoint("long", i));
+        doc.add(new StringField("string", "value", Store.NO));
```

Comment:
Since this query is supposed to match documents that have doc values, we need to index doc values too.

```suggestion
        doc.add(new NumericDocValuesfield("long", i));
        doc.add(new StringField("string", "value", Store.NO));
        doc.add(new SortedDocValuesField("string", new BytesRef("value"));
```

Reply:
Thanks, this makes sense, but why isn't the test failing without those?

Reply:
Year this made me wonder if we should update your new count impl to do this:

```
if (fieldInfo == null || fieldInfo.getDocValuesType() == DocValuesType.NONE) {
  return 0; // the field doesn't index doc values
}
```

Reply:
Indeed, with this change the doc values are needed.